dia_tap_xero/credentials.py

```diff
@@ -13,61 +13,66 @@
-    SSM.put_parameter(
-        Name="/airflow/tap_xero_refresh_token",
-        Overwrite=True,
-        Value=json_response["refresh_token"],
-        Type="SecureString",
-    )
-
-    return config
-
-
-def refresh_tokens(config):
-    token_refresh_url = "https://identity.xero.com/connect/token"
-    b64_id_secret = base64.b64encode(
-        bytes(config["client_id"] + ":" + config["client_secret"], "utf-8")
-    ).decode("utf-8")
-    response = requests.post(
-        token_refresh_url,
-        headers={
-            "Authorization": "Basic " + b64_id_secret,
-            "Content-Type": "application/x-www-form-urlencoded",
-        },
-        data={"grant_type": "refresh_token", "refresh_token": config["refresh_token"]},
-    )
-    json_response = response.json()
-
-    config = rotate_refresh_tokens(config, json_response)
-
-    LOGGER.info("Credentials refreshed, new tokens saved to config")
-    return config
+class XeroCredentials(object):
+    def __init__(self, config):
+        self.SSM = boto3.client("ssm", region_name=config["region_name"])
+        self.config = config
+
+    def build_oauth_headers(self):
+        # Get refresh token from AWS SSM
+        parameter = self.SSM.get_parameter(
+            Name="/airflow/tap_xero_refresh_token", WithDecryption=True
+        )
+
+        self.config["refresh_token"] = parameter["Parameter"]["Value"]
+
+        self.config = self.refresh_tokens()
+
+        oauth_headers = {
+            "Authorization": "Bearer " + self.config["access_token"],
+            "Xero-tenant-id": self.config["tenant_id"],
+            "Accept": "application/json",
+        }
+
+        return oauth_headers
+
+    def rotate_refresh_tokens(self, json_response):
+
+        self.config["refresh_token"] = json_response["refresh_token"]
+        self.config["access_token"] = json_response["access_token"]
+
+        # Rotate refresh token on AWS SSM
+        self.SSM.put_parameter(
+            Name="/airflow/tap_xero_refresh_token",
+            Overwrite=True,
+            Value=json_response["refresh_token"],
+            Type="SecureString",
+        )
+
+        return self.config
+
+    def refresh_tokens(self):
+        token_refresh_url = "https://identity.xero.com/connect/token"
+        b64_id_secret = base64.b64encode(
+            bytes(
+                self.config["client_id"] + ":" + self.config["client_secret"], "utf-8"
+            )
+        ).decode("utf-8")
+        response = requests.post(
+            token_refresh_url,
+            headers={
+                "Authorization": "Basic " + b64_id_secret,
+                "Content-Type": "application/x-www-form-urlencoded",
+            },
+            data={
+                "grant_type": "refresh_token",
+                "refresh_token": self.config["refresh_token"],
+            },
+        )
+        json_response = response.json()
+
+        self.config = self.rotate_refresh_tokens(json_response)
```

Comment:
is the assignment to `self.config` required here? `rotate_refresh_tokens` already updates the `refresh_token` and `access_token` so I think this is reassigning the same values

Reply:
ah good point. this was old from before it was a class